sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/Crawler.scala

```diff
@@ -130,55 +135,101 @@ class Crawler extends CliTool {
   //TODO: Robots.txt
 
   override def run(): Unit = {
-
     //STEP : Initialize environment
     init()
 
     val solrc = this.job.newCrawlDbSolrClient()
+    LOG.info("Committing crawldb..")
+    solrc.commitCrawlDb()
```

Comment:
I don't think committing before starting the job is necessary. Please explain?

Reply:
This is a part of a bigger change that was made to sparkler in memex, where documents were streamed to solr as they were fetched and parsed. Since those two steps happen in memory and if any failure occurs we lose the most expensive part of crawling, fetching. 
I would recommend persisting to solr and introducing more "STATUS" fields that indicate whether a page has been "FETCHED, "PARSED" and "SCORED". There could be more, but these are on top of my mind.